fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/TransactionManagerImplTest.java

```diff
@@ -60,6 +60,12 @@ public void testGetTransaction() {
 
     @Test(expected = RuntimeException.class)
     public void testGetTransactionWithInvalidID() {
-        final TransactionImpl tx = (TransactionImpl) testTxManager.get("invalid-id");
+        testTxManager.get("invalid-id");
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testGetExpiredTransaction() {
+        testTx.expire();
+        testTxManager.get(testTx.getId());
```

Comment:
Hmm, there's actually a bug here. The runtime exception you're getting is an NPE caused by like 91 here:
https://github.com/fcrepo4/fcrepo4/pull/1586/files#diff-1704d54771cc04a4b70dddfed086aba7R91

The persistent session is null, so it can't call rollback. It'd be helpful to verify that rollback gets called on the persistent session in this test. Also, expecting a `RepositoryRuntimeException` instead of RuntimeException would catch this particular bug (although a more specific exception would be better).